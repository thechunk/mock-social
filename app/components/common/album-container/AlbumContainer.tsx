import * as React from 'react';
import {PureComponent} from 'react';
import WrappedWithAlbums from "./WrappedWithAlbums";


export default class AlbumContainer extends PureComponent<IAlbumContainer> {
    render() {
        return (
            <WrappedWithAlbums
                id={this.props.userId}
                limit={this.props.limit}
                withAlbums={this.props.withAlbums}
                renderItem={this.props.renderItem} />
        )
    }
}
