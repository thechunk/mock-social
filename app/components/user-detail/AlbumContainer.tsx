import * as React from 'react';
import {PureComponent} from 'react';
import CopyToChildren from '../utility/CopyToChildren';
import withUserAlbumsData from '../../hoc/withUserAlbumsData';
import withUserPhotosData from '../../hoc/withAlbumPhotosData';

interface IAlbumData {
    limit: number
}
class AlbumData extends PureComponent<IAlbumData & IWithUserAlbumsDataProps> {
    displayAlbums(albums: Array<IAlbum>): Array<IAlbum> {
        return albums.sort((a, b) => b.id - a.id).slice(0, this.props.limit);
    }

    render() {
        const displayAlbum = this.displayAlbums(this.props.albums);
        return displayAlbum.map(v => (
            <CopyToChildren key={v.id} object={{id: v.id}}>{this.props.children}</CopyToChildren>
        ))
    }
}

class PhotoData extends PureComponent<IWithUserPhotosDataProps> {
    static displayPhoto(photos: Array<IPhoto>): IPhoto | undefined {
        return photos.sort((a, b) => a.id - b.id).pop();
    }

    render() {
        const displayPhoto = PhotoData.displayPhoto(this.props.photos);
        return (
            <CopyToChildren object={{source: {uri: displayPhoto ? displayPhoto.url : null}}}>
                {this.props.children}
            </CopyToChildren>
        );
    }
}

const WrappedWithAlbumData = withUserAlbumsData(AlbumData);
const WrappedWithPhotoData = withUserPhotosData(PhotoData);
export default class AlbumContainer extends PureComponent<IAlbumContainer> {
    render() {
        return (
            <WrappedWithAlbumData id={this.props.userId} limit={this.props.limit}>
                <WrappedWithPhotoData>{this.props.children}</WrappedWithPhotoData>
            </WrappedWithAlbumData>
        )
    }
}
