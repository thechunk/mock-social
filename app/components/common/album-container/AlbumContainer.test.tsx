import * as React from 'react';
import {shallow} from 'enzyme';
import AlbumContainer from './AlbumContainer';
import {Text} from 'react-native';

describe('AlbumContainer', () => {
    it('renders correctly', () => {
        expect(shallow(<AlbumContainer userId={0} limit={1} renderItem={v => <Text>{v.url}</Text>}/>)).toMatchSnapshot();
    });
});