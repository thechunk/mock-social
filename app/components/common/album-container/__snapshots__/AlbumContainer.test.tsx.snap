// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumContainer renders correctly 1`] = `
<Component
  id={0}
  limit={1}
  renderItem={[Function]}
/>
`;
